pontoon/translate/static/js/translate.es6

```diff
@@ -0,0 +1,499 @@
+/**
+ * Model for Entities, which represent a single string to be translated.
+ */
+class Entity {
+  constructor(data) {
+    $.extend(this, data);
+  }
+
+  /**
+   * Return the first approved translation in this.translations, or null
+   * if there are none.
+   */
+  get approvedTranslation() {
+    for (let translation of this.translations) {
+      if (translation.approved) {
+        return translation;
+      }
+    }
+
+    return null;
+  }
+
+  /**
+   * Return the current translation status.
+   */
+  get status() {
+    if (this.approvedTranslation) {
+      return 'approved';
+    } else if (this.translations.some((t) => t.fuzzy)) {
+      return 'fuzzy'
+    } else if (this.translations.length > 0) {
+      return 'translated';
+    } else {
+      return '';
+    }
+  }
+
+  updateTranslation(string, localeCode) {
+    let url = `/entity/${this.pk}/translations/${localeCode}/`;
+    return $.post(url, {string: string}).then((translations) => {
```

Comment:
There are two ES6 tricks that you could do here. 

1. `{string: string}` is exactly equivalent to the shorter `{string}`. In particular, if the name of key and the variable are the same, you can just say it once.
2. `(translations) => {` is exactly equivalent to `translations => {` (since there is only one simple argument). I actually don't know if dropping the parens makes me happy or not, but it is an option.